credible/credible-layer-overview.mdx

```diff
@@ -0,0 +1,198 @@
+        actualBalance0Decrease >= expectedBalance0Decrease - tolerance0,
+        "Token0 active balance decrease not proportional to shares burned"
+    );
+}
+```
+
+<Note>This code is truncated for simplicity; read on for [the full explanation on how the Bunni hack](/assertions-book/previous-hacks/bunni-xyz-rounding-error.mdx) could have been prevented with an assertion.</Note>
+
+A proportionality assertion that ensures balance decreases match the share burns would have prevented this $8.4M loss in September 2025.
+
+Now that we understand how invariants are selected and written as assertions, let's dive into how the network node decides to include or drop a transaction.
+
+## Network Enforcement of Assertions
+
+Now that we have our assertion and the target contract that triggers it, we have to let the network sidecar know the contract should be protected by the provided assertion. We call this submitting the assertion for enforcement.
+
+As seen in the diagram below, the dapp submits the assertions to the Credible Layer Smart Contracts. These smart contracts are the source of truth and supply the sidecar with the relevant assertion when a transaction interacts with a protected contract:
+
+![Submitting assertions for enforcement](/images/overview-step-2.png)
+
+### How Enforcement Works
+
+A network node running Phylax's sidecar system does not care _how_ a transaction violates the protocol's rule. All that matters is _whether_ the [assertion](/credible/glossary#assertion) returns true or false after the transaction. If the assertion returns false, the node drops the transaction before it can be included in a block.
+
+You can read the full technical flow in [How It Works](/credible/how-it-works).
+
+To illustrate it better, when a user opens a position on a yield protocol, the transaction touches multiple contracts:
+
+![A single transaction can interact with multiple contracts](/images/phylax-multi-contract-interaction.png)
+
+Each contract that has assertions is checked by the sidecar, which simulates the cumulative end states in the PhEVM:
+
+![Each contract with assertions is checked by the sidecar](/images/phylax-assertion-checking.png)
+
+### Assertion Validation
+
+Once the transaction is sent by the user, it is seen by the network node. The node then passes it to the sidecar to check if the transaction's final state would violate an [assertion](/credible/glossary#assertion) in a separate replica environment called the [PhEVM](/credible/glossary#phevm-phylax-evm). When completed, the sidecar returns the result to the node so it can include or drop the transaction. This validation process never exceeds the time to produce a block.
+
+In other words, the transaction will be included into the block only if every assertion across all touched contracts evaluates to true. If even one assertion fails, the network node will see this failed assertion and exclude it from finalization.
```

Comment:
I would add a paragraph here saying that the exact architecture depends on the stack (e.g op stack, ethereum mainnet, etc.) and give the example fo op stack that is implemented (the validation) as part of an external builder that integrates via rollup-boost.

The main takeaway here is that the core of the system/idea remains the same and the exact integration depends on the underlying technology. That would be probably one of the main questions the reader has as they are reading thid